### PR TITLE
Make defaultRevision BoolValue instead of raw bool

### PR DESCRIPTION
**Pull request by Monkeyanator**
Need to be able to distinguish between unset `defaultRevision` and user explicitly setting `defaultRevision: false`

### CODE REVIEW

File: operator/v1alpha1/operator.proto [hunk 1]
```diff
@@ -97,7 +99,8 @@ message IstioOperatorSpec {
 
     // Identify whether this revision is the default revision for the cluster
     // This option is currently experimental.
-    bool default_revision = 17;
+    google.protobuf.BoolValue default_revision = 17;
```
Comment (costinm) [inline operator/v1alpha1/operator.proto:102]: That's still pretty inconvenient. Bools are bad, we should avoid them.

Also 'default_revision' is not ideal - can we turn this into an enum ? Can have 2 values for start, but I suspect we may have others in future. 

Comments are not clear - what does it mean for a user that it's a 'default revision' ?

Comment (Monkeyanator) [reply operator/v1alpha1/operator.proto:102]: Other than `DEFAULT` and `NOT DEFAULT` what other types of revisions could there be? Agreed an enum would be better but having a field like `revisionType: DEFAULT` or `revisionType: NOT DEFAULT` feels awkward.

The comments are outdated, I'll update.

**Comment by costinm** [comment]
I would also avoid using the term 'revision' in the API.

With string:
installType: "" ( default value is empty )
installType: "default"

In future we may refine it - for example add "injection" for an install
that will be used for injection only,
or "ca" for an install that will be used for CA only ( while Istiod has all
the functions, it can also be
deployed in a micro-services model ).

File: operator/v1alpha1/operator.proto (hunk 1 above)
Comment (costinm) [reply operator/v1alpha1/operator.proto:102]: It's not going to be 'NOT_DEFAULT' - just empty ( default value ).

install_type: default  

In future we may have 'inject_only' or some other modes of installing.

Comment (hzxuzhonghu) [reply operator/v1alpha1/operator.proto:102]: How about make it string, if unset, default it to the version installed

**Pull request closed without merging**